### PR TITLE
use safe allocator in hnsw implementation

### PR DESCRIPTION
**Pull request by wxyucs**

### CODE REVIEW

File: src/index/hnsw.cpp [hunk 1]
```diff
@@ -77,7 +77,7 @@ HNSW::HNSW(std::shared_ptr<hnswlib::SpaceInterface> space_interface,
         alg_hnsw =
             std::make_shared<hnswlib::HierarchicalNSW>(space.get(),
                                                        DEFAULT_MAX_ELEMENT,
-                                                       allocator,
+                                                       allocator_,
```
Comment (jiaweizone) [inline src/index/hnsw.cpp:80]: Missing changes?

File: src/index/hnsw.h [hunk 2]
```diff
@@ -77,6 +77,7 @@ class HNSW : public Index {
          Allocator* allocator = nullptr);
 
     virtual ~HNSW() {
+        alg_hnsw = nullptr;
```
Comment (jiaweizone) [inline src/index/hnsw.h:80]: alg_hnsw -> alg_hnsw_

File: src/index/hnsw.cpp (hunk 1 above)
Comment (wxyucs) [reply src/index/hnsw.cpp:80]: yes, missing in last commit

**Review by inabao — APPROVED** [review]

File: src/index/hnsw.h (hunk 2 above)
Comment (jiaweizone) [reply src/index/hnsw.h:80]: > alg_hnsw -> alg_hnsw_

will change in next PR

**Review by jiaweizone — APPROVED** [review]
LGTM

**Pull request merged**